hw14.1_todo_list/src/index.ts

```diff
@@ -0,0 +1,194 @@
+// Вам необхідно написати додаток Todo list. 
+//У списку нотаток повинні бути методи для додавання нового запису, видалення, редагування та отримання повної інформації про нотатку за ідентифікатором, а так само отримання списку всіх нотаток. 
+//Крім цього, у користувача має бути можливість позначити нотаток, як виконаний, і отримання інформації про те, скільки всього нотаток у списку і скільки залишилося невиконаними. 
+//Нотатки не повинні бути порожніми.
+// Кожний нотаток має назву, зміст, дату створення і редагування та статус. Нотатки бувають двох типів. Дефолтні та такі, які вимагають підтвердження при ридагуванні.
+// Окремо необхідно розширити поведінку списку та додати можливість пошуку нотатка за ім'ям або змістом.
+// Також окремо необхідно розширити список можливістю сортування нотаток за статусом або часом створення.
+
+enum StatusTodoListEnum {
+  New = 'NEW',
+  IsPerformed = 'IS PERFORMED',
+  Executed = 'EXECUTED'
+}
+
+enum TypeOfTaskEnum {
+  Default = 'DEFAULT',
+  EditConfirm = 'EDIT CONFIRMATION'
+}
+
+interface ITodoTask {
+  id: number;
+  name: string;
+  description: string;
+  dateCreated: Date;
+  dateEdited?: Date | null;
+  editing: boolean;
+  status: StatusTodoListEnum;
+  typeOfTask: TypeOfTaskEnum;
+}
+
+interface ITodoList {
+  addNewTask(task: ITodoTask): void;
+  removeTask(taskId: number): void;
+  editingTask(taskId: number, updatedTask: Partial<ITodoTask>): void;
+  getInformationTask(taskId: number): ITodoTask | undefined;
+  getAllOfTasks(): ITodoTask[];
+  getTotalTasksCount(): number;
+  markTaskAsExecuted(taskId: number): void;
+  getExecutedTaskCount(): number;
+  getTotalTasksCount(): number;
+  searchTasks(query: string): ITodoTask[];
+  sortTasks(by: 'status' | 'dateCreated'): ITodoTask[]
+}
+
+class TodoList implements ITodoList {
+  private tasks: ITodoTask[] = [];
+  private nextId: number = 1;
+
+  
+  //Add a new task
+  public addNewTask(task: ITodoTask): void {
```

Comment:
There is a requirement for non-empty tasks, but I can't find any checks for this.